both/_collections.js

```diff
@@ -286,6 +286,18 @@ Registrations.attachSchema({
   },
   uniqueName: {
     type: String,
+    unique: true,
+    custom: function(a) {
+      var compId = this.field('competitionId').value;
+      var uniqueMatch = Registrations.find({
```

Comment:
For efficiency, let's just do a fineOne here instead. I believe that works, right?